### ISSUE TITLE
match assembly operand labels and order in INSTR_CREATE_ macros for ARM

### ISSUE THREAD

**Issue by derekbruening** [DynamoRIO/dynamorio#1563]
_From [bruen...@google.com](https://code.google.com/u/109494838902877177630/) on November 17, 2014 17:42:25_

The first batch of INSTR_CREATE_ macros for ARM are going to differ from assembly in at least the following ways:
- Dsts are always first, while ARM asm has memory dsts later in the order
- The macros are using Rd for load dsts where asm uses Rt
- The macros are using Vn for single source operands for easy doxygen generation where asm uses Vm
- The macros are using Vd and Rd for OP_vmov and OP_vdup dsts where asm uses Vm and Rt
- The macros are using Vt and Vm for the srcs of OP_vcmp where asm uses Vd and Vm

Basically, the asm conventions have violations and exceptions that make them hard to program toward.

_Original issue: http://code.google.com/p/dynamorio/issues/detail?id=1563_
